api/src/Controller/Event/BaseEvent.php

```diff
@@ -117,5 +118,59 @@ public function __construct(Container $container)
     }
 
 
+    protected function buildPutPostBody($request, $response, $params, $required, $event)
+    {
+        $body = $this->checkRequiredBody($request, $required);
+        unset($body['cycle']);
+
+        if ($event) {
+            $cycle_from = $event['cycle'];
+            $cycle_to = $event['cycle'];
+        } else {
+            $cycle_from = null;
+            $cycle_to = null;
+        }
+        if (array_key_exists('date_from', $body)) {
+            try {
```

Comment:
Um...not actually seeing any difference to this code...

Reply:
Oh I missed your point. I thought you meant the blocks between the PUT and POST methods which I did combine.  Not the methods WITHIN the function.  I can take another pass later today if you feel it worth it.

Reply:
Oh, yeah, sorry. Refactor all of that inner logic for handling date to/from to a method, please :-D

Reply:
Ok fixed.